Mentioned non_field_errors in form validation docs. (#12655)
django/django: docs/ref/forms/validation.txt

In this code, if the validation error is raised, the form will display an
error message at the top of the form (normally) describing the problem.

> Note that errors generated here will be form errors and not field errors,
> to display these errors you'll have to define ``{{ form.non_field_errors }}``
> in your template.

Reviewer: I edited down but don't think I lost any information:

```suggestion
Such errors are non-field errors, which are displayed in the template with ``{{ form.non_field_errors }}``.
```

Author: Thanks @adamchainz, this is much better. :)
